### PR TITLE
AWS SDK v2 support

### PR DESCRIPTION
**Pull request by Kirik0**
Performed local testing using GETs, PUTs, DELETEs. Performed a couple secret rotations, pulled credential history, updated metadata.

### CODE REVIEW

File: fidelius-sdk/src/main/java/org/finra/fidelius/FideliusClient.java [hunk 1]
```diff
@@ -17,114 +17,128 @@
...
+            kmsBuilder = kmsBuilder.region(awsRegion);
+            stsBuilder = stsBuilder.region(awsRegion);
+            clientBuilder = clientBuilder.region(awsRegion);
+            lambdaClientBuilder = lambdaClientBuilder.region(awsRegion);
         }
-        lambda = lambdaClientBuilder.build();
-        client = clientBuilder.build();
-        awsSecurityTokenService = stsBuilder.build();
-        jCredStash = new JCredStash(ddbBuilder.build(), kmsBuilder.build(), awsSecurityTokenService);
+        if(envConfig.hasProxyEnv()) {
```
Comment (Brijeshrpatel9) [inline fidelius-sdk/src/main/java/org/finra/fidelius/FideliusClient.java:119]: Not sure if I understand this if condition, don't we have to initialize all the clients if proxy env is provided and if not provided, like if.....else....

File: fidelius-sdk/src/main/java/org/finra/fidelius/JCredStash.java [hunk 2]
```diff
@@ -49,87 +38,109 @@
...
 public class JCredStash {
-    protected AmazonDynamoDB amazonDynamoDBClient;
-    protected AWSKMS awskmsClient;
+    protected DynamoDbClient dynamoDbClient;
+    protected KmsClient kmsClient;
     protected CredStashCrypto cryptoImpl;
-    protected AWSSecurityTokenService awsSecurityTokenService;
-    protected DynamoDB dynamoDB;
+    protected StsClient stsClient;
+    protected static  List<String> TABLE_HEADERS = Arrays.asList("name", "component", "sdlc", "contents", "version", "updatedBy", "updatedOn", "key", "hmac", "source", "sourceType");
```
Comment (Brijeshrpatel9) [inline fidelius-sdk/src/main/java/org/finra/fidelius/JCredStash.java:45]: Unused variable. You can remove it

File: fidelius-service/src/main/java/org/finra/fidelius/services/aws/DynamoDBService.java [hunk 3]
```diff
@@ -88,29 +86,6 @@ public <T>List<T> queryDynamoDB(DynamoDBQueryExpression queryRequest, Class<T> c
             logger.info(String.format("Query completed in %.3f seconds", (System.currentTimeMillis() - startTime) / 1000.0));
         }
 
-        return queryResults;
-    }
-
-    // Creates a new DynamoDBMapper object
-    public DynamoDBMapper createMapper(String account, String region, String tableName) {
```
Comment (Brijeshrpatel9) [inline fidelius-service/src/main/java/org/finra/fidelius/services/aws/DynamoDBService.java:95]: We don't need mapping?

File: fidelius-sdk/src/main/java/org/finra/fidelius/JCredStash.java (hunk 2 above)
Comment (Kirik0) [reply fidelius-sdk/src/main/java/org/finra/fidelius/JCredStash.java:45]: Good catch. Will remove.

File: fidelius-service/src/main/java/org/finra/fidelius/services/aws/DynamoDBService.java (hunk 3 above)
Comment (Kirik0) [reply fidelius-service/src/main/java/org/finra/fidelius/services/aws/DynamoDBService.java:95]: DynamoDBMapper is no longer supported in SDK v2. Needed to do some workarounds to keep the logic as is.

File: fidelius-sdk/src/main/java/org/finra/fidelius/FideliusClient.java (hunk 1 above)
Comment (Kirik0) [reply fidelius-sdk/src/main/java/org/finra/fidelius/FideliusClient.java:119]: We're still building the Clients on lines 128-131. For AWS SDK v2, client builders like the DynamoDbClientBuilder need an additional HTTP Client built-in for proxy configuration to work. The if-block there takes care of that if proxy configuration is provided but skips if it's not.

**Review by SMxJrz — CHANGES_REQUESTED** [review]

**Review by SMxJrz — APPROVED** [review]

**Pull request merged**